### PR TITLE
espanso: 2.1.8 -> 2.2.1

### PR DESCRIPTION
**Pull request by need-4-sleep**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pepega-hacker** [comment]
Hi @eclairevoyant, I'm really sorry for the ping; could you please review this? This PR also closes #286384.

**Comment by kirillrdy** [comment]
@pepega-hacker can you remove merge commit

**Comment by pepega-hacker** [comment]
Uh what

**Comment by kirillrdy** [comment]
github says 0 commits,

you can push to same branch and reopen this PR

**Comment by pepega-hacker** [comment]
Whew 😅 thank you

**Comment by kirillrdy** [comment]
@pepega-hacker darwin build failed, let me know if you need assistance fixing it

File: pkgs/applications/office/espanso/default.nix [hunk 1]
```diff
@@ -55,10 +55,6 @@ rustPlatform.buildRustPackage rec {
     };
   };
 
-  cargoPatches = lib.optionals stdenv.isDarwin [
```
Comment (pyrox0) [inline pkgs/applications/office/espanso/default.nix:58]: Is this the reason that the build fails on darwin? I noticed that this was only originally tested on x86_64-linux, but this is a darwin patch, so this should have been built or tested on darwin without this before commiting. Could someone test if re-adding this fixes the darwin build? I do not have a darwin machine myself so I cannot test this.

Comment (kirillrdy) [reply pkgs/applications/office/espanso/default.nix:58]:
```
>   ld: framework not found System
       >   clang-16: error: linker command failed with exit code 1 (use -v to see invocation)
       >   make[1]: *** [Makefile:117: wxrc] Error 1
```

Comment (kirillrdy) [reply pkgs/applications/office/espanso/default.nix:58]: yeah i think its this patch, I can test later today

**Comment by pepega-hacker** [comment]
Sorry about that. When i reopened the PR earlier i just copied the package from my home flake repo and forgot i removed that line.

**Comment by kirillrdy** [comment]
no worries, that's what CI is for !

**Comment by n8henrie** [comment]
If you're interested, I have a version working on Darwin here: https://github.com/n8henrie/nixpkgs/blob/espanso-darwin-aarch64/pkgs/applications/office/espanso/default.nix

I was surprised to find that I didn't need the inject-wx-on-darwin patch at all, works fine for me without it.

Will also make a few other notes

**Comment by n8henrie** [comment]
I don't know if this is kosher by nixpkgs guidelines, but [this commit](https://github.com/espanso/espanso/commit/471c2841977aa650a08a786679adf2697706c69b) is subsequent to the most recent tagged version (2.2.1, targeted by this commit), but it has a (very small) fix that resolves a runtime crash on aarch64-darwin and would be tremendously helpful to have packaged.

Unfortunately there is no definite timeframe for the next release (https://github.com/espanso/espanso/pull/1946#issuecomment-2054211453, https://github.com/espanso/espanso/issues/1928), so it's unclear when a new named version with this fix would appear.

Alternatively, sticking with 2.2.1 but including this patch should work:

```diff
diff --git a/espanso-detect/src/mac/native.mm b/espanso-detect/src/mac/native.mm
index b96d7ac..aa7bcaa 100644
--- a/espanso-detect/src/mac/native.mm
+++ b/espanso-detect/src/mac/native.mm
@@ -137,6 +137,7 @@
         }
     }];
   });
+  return nullptr;
 }
 
 OSStatus hotkey_event_handler(EventHandlerCallRef _next, EventRef evt, void *userData)
```

Could this be considered? Either of the above should close https://github.com/NixOS/nixpkgs/issues/247162

**Comment by n8henrie** [comment]
I'd also happily join the list of espanso maintainers if OK by @kimat @thehedgeh0g

**Comment by malt3** [comment]
I updated espanso to a pre-release version here to fix a wayland issue: #311724

**Pull request closed without merging**